app/main.py

```diff
@@ -1,23 +1,41 @@
 import init_django_orm  # noqa: F401
-
-from django.db.models import QuerySet
+from db.models import LoyaltyProgram, LoyaltyProgramParticipant
+from django.db.models import F, Q, QuerySet
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    queryset = LoyaltyProgram.objects.filter().values_list("name",
+                                                           "bonus_percentage")
+    return queryset
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    queryset = LoyaltyProgramParticipant.objects.filter(
+        last_activity__lt="2022-01-01",
+        last_activity__gt="2021-01-01"
+    ).values("customer__first_name")
+    return queryset
 
 
 def most_active_customers() -> QuerySet:
-    pass
+    queryset = LoyaltyProgramParticipant.objects.\
+        filter().values_list("customer__first_name",
+                             "customer__last_name",
+                             "sum_of_spent_money"
+                             ).order_by("-sum_of_spent_money")[:5]
+    return queryset
 
 
 def clients_with_i_and_o() -> QuerySet:
-    pass
+    queryset = LoyaltyProgramParticipant. \
+        objects.filter(
+            Q(customer__first_name__startswith="I") | Q
```

Comment:
I think it's better to split lines before the `|` operator. It seems more readable to me that way.